Library/Homebrew/extend/ARGV.rb

```diff
@@ -99,7 +99,12 @@ def homebrew_developer?
   end
 
   def sandbox?
-    include?("--sandbox") || !ENV["HOMEBREW_SANDBOX"].nil?
+    enable = include?("--sandbox") || !ENV["HOMEBREW_SANDBOX"].nil?
+    autodisable = interactive? || debug?
+    if enable && autodisable
+      opoo "The sandbox cannot be used in debug or interactive mode."
+    end
+    enable && !autodisable
```

Comment:
Looking at the other functions in this file I think probably all of this code should be moved to a utility function elsewhere and the method left as it was.